src/scripts/main.js

```diff
@@ -1,3 +1,349 @@
+const cell14 = document.getElementById('1_4');
+
+const cell21 = document.getElementById('2_1');
+const cell22 = document.getElementById('2_2');
+const cell23 = document.getElementById('2_3');
+const cell24 = document.getElementById('2_4');
+
+const cell31 = document.getElementById('3_1');
+const cell32 = document.getElementById('3_2');
+const cell33 = document.getElementById('3_3');
+const cell34 = document.getElementById('3_4');
+
+const cell41 = document.getElementById('4_1');
+const cell42 = document.getElementById('4_2');
+const cell43 = document.getElementById('4_3');
+const cell44 = document.getElementById('4_4');
+
+const messageStart = document.querySelector('.message-start');
+const messageLose = document.querySelector('.message-lose');
+const messageWin = document.querySelector('.message-win');
+
+const score = document.querySelector('.game-score');
+let currentScore = 0;
+
+score.textContent = currentScore;
+
+const board = [
+  cell11, cell12, cell13, cell14,
+  cell21, cell22, cell23, cell24,
+  cell31, cell32, cell33, cell34,
+  cell41, cell42, cell43, cell44,
+];
+
+document.querySelector('button').addEventListener('click', buttons);
+document.addEventListener('keydown', arrow => arrowTracking(arrow));
+
+function buttons() {
+  const button = document.querySelector('button');
+
+  if (button.textContent === 'Start') {
+    button.classList.remove('start');
+    button.classList.add('restart');
+    button.textContent = 'restart';
+
+    messageStart.classList.add('hidden');
+
+    addRandomCell();
+    addRandomCell();
+  } else {
+    score.textContent = 0;
+
+    board.forEach(cell => {
+      cell.classList = ['field-cell'];
+      cell.textContent = '';
+    });
+
+    button.classList.remove('restart');
+    button.classList.add('start');
+    messageWin.classList.add('hidden');
+    messageLose.classList.add('hidden');
+    button.textContent = 'Start';
+    messageStart.classList.remove('hidden');
+  }
+};
+
+function randomCell() {
+  let randomIndex = Math.floor(Math.random() * board.length);
+  let randomElement = board[randomIndex];
+
+  while (randomElement.classList.length >= 2) {
+    randomIndex = Math.floor(Math.random() * board.length);
+    randomElement = board[randomIndex];
+  };
+
+  return randomElement;
+};
+
+function randomNumber() {
+  const randomNumb = Math.floor(Math.random() * 100);
+
+  return randomNumb < 90 ? 2 : 4;
+};
+
+function addRandomCell() {
+  const randomEl = randomCell();
+  const randomNum = randomNumber();
+
+  randomEl.classList.add(`field-cell--${randomNum}`);
+  randomEl.textContent = randomNum;
+};
+
+function arrowTracking(arrow) {
+  const key = arrow.code;
+
+  switch (key) {
+    case 'ArrowUp':
+      processElements('up');
+      break;
+    case 'ArrowRight':
+      processElements('right');
+      break;
+    case 'ArrowDown':
+      processElements('down');
+      break;
+    case 'ArrowLeft':
+      processElements('left');
+      break;
+    default:
+      break;
```

Comment:
Will be better if your create a variable for these words
```suggestion
    case 'ArrowUp':
      processElements('up');
      break;
      
    case 'ArrowRight':
      processElements('right');
      break;
      
    case 'ArrowDown':
      processElements('down');
      break;
      
    case 'ArrowLeft':
      processElements('left');
      break;
      
    default:
      break;
```